### PR TITLE
Notifications + Swift Mk 8: View Lifecycle

### PR DESCRIPTION
**Pull request by jleandroperez**
#### Details:
This PR adds no new functionality. We'll gradually migrate all of the Notifications-Y code over to Swift.
Once it's ready, we'll address code improvements / technical debt.

Reference #5613

--

#### How to Test:
- Please, verify that the Notifications List shows up properly
- Rotate the view and check that it works as expected
- Verify that the filters work as they should
- Try logging out and into a different account
- Try opening an Automattcher Notification. The reader should show up onscreen
- Try opening a Comment-Y Notification, and validate that everything works as expected

@kwonye *ALMOST* there with **NotificationsViewController**.
Ready for review sir!!

Thanks in advance!!

### CODE REVIEW

**Comment by kwonye** [comment]
Tapping "Happy Anniversary with WordPress.com!" resulted in a crash

```
2016-07-26 14:34:21:303 WordPress[4440:151167] Pushing Notification Details for: [2433537555]
2016-07-26 14:34:21.350 WordPress[4440:151167] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSArrayM insertObject:atIndex:]: object cannot be nil'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010f8d2d85 __exceptionPreprocess + 165
	1   libobjc.A.dylib                     0x000000010f346deb objc_exception_throw + 48
	2   CoreFoundation                      0x000000010f793cc5 -[__NSArrayM insertObject:atIndex:] + 901
	3   WordPress                           0x0000000107f247f8 -[SuggestionService updateSuggestionsForSiteID:] + 232
	4   WordPress                           0x0000000107f246c3 -[SuggestionService suggestionsForSiteID:] + 147
	5   WordPress                           0x0000000107c9e6a9 -[SuggestionsTableView initWithSiteID:] + 249
	6   WordPress                           0x0000000107ece967 -[NotificationDetailsViewController setupSuggestionsView] + 135
	7   WordPress                           0x0000000107ecd308 -[NotificationDetailsViewController viewDidLoad] + 168
	8   UIKit                               0x000000010b777984 -[UIViewController loadViewIfRequired] + 1198
	9   WordPress                           0x0000000107ecd881 -[NotificationDetailsViewController setupWithNotification:] + 81
	10  WordPress                           0x00000001082ca762 _TFC9WordPress27NotificationsViewController15prepareForSeguefTCSo17UIStoryboardSegue6senderGSqPs9AnyObject___T_ + 434
	11  WordPress                           0x00000001082cac0b _TToFC9WordPress27NotificationsViewController15prepareForSeguefTCSo17UIStoryboardSegue6senderGSqPs9AnyObject___T_ + 75
	12  UIKit                               0x000000010bd245d5 -[UIStoryboardSegueTemplate _performWithDestinationViewController:sender:] + 369
	13  UIKit                               0x000000010bd24433 -[UIStoryboardSegueTemplate _perform:] + 82
	14  UIKit                               0x000000010b77a5f8 -[UIViewController performSegueWithIdentifier:sender:] + 99
	15  UIKit                               0x000000011e616cc4 -[UIViewControllerAccessibility performSegueWithIdentifier:sender:] + 39
	16  WordPress                           0x00000001082cd9a8 _TFC9WordPress27NotificationsViewController26showDetailsForNotificationfCSo12NotificationT_ + 3576
	17  WordPress                           0x00000001082cdada _TToFC9WordPress27NotificationsViewController26showDetailsForNotificationfCSo12NotificationT_ + 58
	18  WordPress                           0x00000001082ca3b4 _TFC9WordPress27NotificationsViewController9tableViewfTCSo11UITableView23didSelectRowAtIndexPathCSo11NSIndexPath_T_ + 436
	19  WordPress                           0x00000001082ca59f _TToFC9WordPress27NotificationsViewController9tableViewfTCSo11UITableView23didSelectRowAtIndexPathCSo11NSIndexPath_T_ + 79
	20  WordPress                           0x0000000107d3c04c -[WPTableViewHandler tableView:didSelectRowAtIndexPath:] + 124
	21  UIKit                               0x000000010b7241c6 -[UITableView _selectRowAtIndexPath:animated:scrollPosition:notifyDelegate:] + 1887
	22  UIKit                               0x000000010b72441b -[UITableView _userSelectRowAtPendingSelectionIndexPath:] + 388
	23  UIKit                               0x000000010b5e8f62 _runAfterCACommitDeferredBlocks + 317
	24  UIKit                               0x000000010b5fce4c _cleanUpAfterCAFlushAndRunDeferredBlocks + 95
	25  UIKit                               0x000000010b609147 _afterCACommitHandler + 90
	26  CoreFoundation                      0x000000010f7f7c37 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
	27  CoreFoundation                      0x000000010f7f7ba7 __CFRunLoopDoObservers + 391
	28  CoreFoundation                      0x000000010f7ed7fb __CFRunLoopRun + 1147
	29  CoreFoundation                      0x000000010f7ed0f8 CFRunLoopRunSpecific + 488
	30  GraphicsServices                    0x0000000112226ad2 GSEventRunModal + 161
	31  UIKit                               0x000000010b5dcf09 UIApplicationMain + 171
	32  WordPress                           0x0000000107c4e3fe main + 62
	33  libdyld.dylib                       0x00000001106e292d start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by jleandroperez** [comment]
@kwonye wooooow nice catch!!. Great news is: apparently, that crash has been lurking for a while (since it's breaking in an unmodified spot).

Should be good now, thanks sir!!

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.m [hunk 1]
```diff
@@ -1,258 +0,0 @@
...
-#pragma mark ====================================================================================
-
-@implementation NotificationsViewController
-
-- (void)dealloc
-{
-    [[NSNotificationCenter defaultCenter] removeObserver:self];
-}
-
-- (instancetype)initWithCoder:(NSCoder *)aDecoder
```
Comment (kwonye) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.m]: I don't see initWithCoder in the new Swift version. I see bits and pieces of this in the new `viewWillAppear`, but it's not necessarily the same. Can you guarantee the functionality is the same?

Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.m]: Yes sir! i've moved the account observer to viewDidLoad (method `startListeningToAccountNotifications`). As per the rest, pre-initializing them in the var declaration itself.

❤️ Swift!

**Comment by kwonye** [comment]
:shipit:

**Comment by jleandroperez** [comment]
Thank you @kwonye!!!

**Pull request merged**